lib/ldclient-rb/ldclient.rb

```diff
@@ -295,7 +258,7 @@ def secure_mode_hash(context)
     # @return [Boolean] true if the client has been initialized
     #
     def initialized?
-      @config.offline? || @config.use_ldd? || @data_source.initialized?
+      @data_system.data_availability == @data_system.target_availability
```

Comment:
Are we sure these lines are equivalent given our previous discussions about this? I can't remember exactly where we landed on that.